tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -409,16 +410,118 @@ def test__update_collection__OK(self):
         self.assertEqual(200, response.status_code)
 
     def test__update_collection_partial_data__OK(self):
-        collection_id = self.generate_collection(self.session).id
-        metadata = {"name": "A new name, and only a new name"}
+        description = "a description"
+        contact_name = "first last"
+        contact_email = "name@server.domain"
+        links = [
+            {"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"},
+        ]
+        publisher_metadata = {
+            "authors": [{"name": "First Last", "given": "First", "family": "Last"}],
+            "journal": "Journal of Anamolous Results",
+            "published_year": 2022,
+            "published_month": 1,
+        }
+        collection_id = self.generate_collection(
+            self.session,
+            description=description,
+            contact_name=contact_name,
+            contact_email=contact_email,
+            links=links,
+            publisher_metadata=publisher_metadata,
+        ).id
+        new_name = "A new name, and only a new name"
+        metadata = {"name": new_name}
         response = self.app.patch(
             f"/curation/v1/collections/{collection_id}",
             data=json.dumps(metadata),
             headers=self.make_owner_header(),
         )
         self.assertEqual(200, response.status_code)
         response = self.app.get(f"curation/v1/collections/{collection_id}")
-        self.assertEqual(response.json["name"], metadata["name"])
+        self.assertEqual(new_name, response.json["name"])
+        self.assertEqual(description, response.json["description"])
+        self.assertEqual(contact_name, response.json["contact_name"])
+        self.assertEqual(contact_email, response.json["contact_email"])
+        self.assertEqual(links, response.json["links"])
+        self.assertEqual(publisher_metadata, response.json["publisher_metadata"])
+
+    def test__update_collection__links_and_doi_management__OK(self):
+        name = "partial updates test collection"
+        links = [
+            {"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"},
+            {"link_name": "doi", "link_type": "DOI", "link_url": "http://doi.doi"},
+        ]
+        new_links = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+            {"link_name": "new doi", "link_type": "DOI", "link_url": "http://doi.org/10.1016"},  # a real DOI
```

Comment:
mock this out rather than hit the DOI server. Otherwise, this could turn into another flakey API like dropbox..